### PR TITLE
etherscan ratios integration

### PR DESCRIPTION
**Pull request by husobee**
### Summary

Etherscan passthrough caching proxy for gasoracle

### Type of change ( select one )

- [ ] Product feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before submitting this PR:

- [ ] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security / privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: utils/clients/etherscan/client.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+	"github.com/brave-intl/bat-go/utils/clients"
+	"github.com/brave-intl/bat-go/utils/closers"
+	appctx "github.com/brave-intl/bat-go/utils/context"
+	"github.com/gomodule/redigo/redis"
+	"github.com/google/go-querystring/query"
+)
+
+const (
+	gasOracleCacheTTLHours = 1 // How long we consider Redis cached FetchCoinMarkets responses to be valid
```
Comment (nvonpentz) [inline utils/clients/etherscan/client.go:19]: Should update the comment or remove it

File: utils/clients/etherscan/client.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+// GenerateQueryString - implement the QueryStringBody interface
+func (p *gasOracleParams) GenerateQueryString() (url.Values, error) {
+	return query.Values(p)
+}
+
+// GasOracleResponse is the response received from coingecko
+type GasOracleResponse map[string]interface{}
+
+// FetchGasOracle fetches the market data for the top coins
```
Comment (nvonpentz) [inline utils/clients/etherscan/client.go:105]: This comment too

File: utils/clients/etherscan/client.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+type baseParams struct {
+	APIKey string `url:"apikey"`
+}
+
+// gasOracleParams for fetching gas oracle
+type gasOracleParams struct {
+	baseParams
+	Module string `url:"gastracker"`
+	Action string `url:"gasoracle"`
+}
```
Comment (nvonpentz) [inline utils/clients/etherscan/client.go:95]: Should this be
```
type gasOracleParams struct {
	baseParams
	Module string `url:"module"`
	Action string `url:"action"`
}
```
?

File: utils/clients/etherscan/client.go [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+// GenerateQueryString - implement the QueryStringBody interface
+func (p *gasOracleParams) GenerateQueryString() (url.Values, error) {
+	return query.Values(p)
+}
+
+// GasOracleResponse is the response received from coingecko
+type GasOracleResponse map[string]interface{}
+
+// FetchGasOracle fetches the market data for the top coins
+func (c *HTTPClient) FetchGasOracle(
```
Comment (nvonpentz) [inline utils/clients/etherscan/client.go:106]: I'll get confirmation, but I think we can drop support for passing through to this gas oracle endpoint.  I think we just need these two:

* `module=gastracker&action=gasestimate` - https://github.com/brave/brave-core/blob/7040d3f232e9d269eb734f4c007d98699584a9b3/components/brave_wallet/browser/asset_ratio_service.cc#L236-L253
* `module=token&action=tokeninfo` - https://github.com/brave/brave-core/blob/7040d3f232e9d269eb734f4c007d98699584a9b3/components/brave_wallet/browser/asset_ratio_service.cc#L275-L292

**Comment by husobee** [comment]
closing for now, but keeping branch.

**Pull request closed without merging**